.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+# Coding format checker and test runner.
+name: Automated tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  lint:
+    name: Linter
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fetch source
+        uses: actions/checkout@v2
+
+      - name: Lint
+        run: cargo fmt --check
+
+  test:
+    name: Test runner
+    runs-on: ubuntu-latest
+
+    steps:
+      # TODO: actions/cache@v2?
```

Comment:
Yes, probably. If you view the logs with timestamps on, most of the time is spent fetching the crates.io index. The actions cache docs have examples of how to add cache to a rust project. Should be fine to copy paste that.

Reply:
That said, the cache only lasts for a week, so you will only feel the benefit from the second push to each pr, and a minute is pretty quick for a CI build time. Let's leave caching for another time.